src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -924,6 +932,38 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
     return false;
 }
 
+// Routine Description:
+// - OSC 9 ; 4 ; state ; progress ST
+//       state: the state to set the taskbar progress indicator to
+//       progress: the progress value to set the taskbar progress indicator with
+// - Parses out the 'state' and 'progress' parameters from the OSC string so
+//   that we can use them to set the taskbar progress indicator
+// - This follows the ConEmu style, more details here:
+//   https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+// Arguments:
+// - string: the string to parse
+// - state: where to store the state value once we parse it
+// - progress: where to store the progress value once we parse it
+// Return Value:
+// - true if we successfully parsed the string, false otherwise
+bool OutputStateMachineEngine::_GetTaskbarProgress(const std::wstring_view string,
+                                                   size_t& state,
+                                                   size_t& progress) const
+{
+    const auto parts = Utils::SplitString(string, L';');
+    if (parts.size() != 3 || std::stoi(til::u16u8(parts.at(0))) != 4)
+    {
+        return false;
+    }
+    state = std::stoi(til::u16u8(parts.at(1)));
+    progress = std::stoi(til::u16u8(parts.at(2)));
+    if (state < 0 || state > 4 || progress < 0 || progress > 100)

```

Comment:
@zadjii-msft thoughts on the validation here for reasonable values? I personally think its a little out of place and should just be done in `IslandWindow.cpp` when we actually use the values

Reply:
I think we're going with the validation in the dispatcher these days? Check out `AdaptDispatch::EraseInDisplay`, that one ensures that the parameters passed to it are sensible, before trying to actually dispatch the action.

(for the record, the parameter validation _used to_ be done in this file, up till like a month ago)

Reply:
On the subject of validation, ideally we should be matching the ConEmu behaviour for the various edge cases and "error" conditions. 

1. Both the _st_ and _pr_ parameters should default to 0, so if those parameters are left off, it should be the same as sending an explicit "remove progress" command. So both of the following sequences should reset the progress bar.

        OSC 9;4 ST
        OSC 9;4;1 ST

2. As with most VT sequences, unexpected additional parameters should be ignored, so the sequence below should still successfully set the progress to 50%.

        OSC 9;4;1;50;123 ST

3. Out of range _pr_ values should be clamped, so the sequence below should still successfully set the progress to 100%.

        OSC 9;4;1;999 ST

It's not critical if we don't get all of these edge cases right, but the closer we match their implementation, the better it is for interoperability.